slsim/lens.py

```diff
@@ -460,20 +444,24 @@ def image_observer_times(self, t_obs):
         :rtype: numpy array. Each element of the array corresponds to different image
             observation times.
         """
-        arrival_times = self.point_source_arrival_times()
-        if type(t_obs) is np.ndarray and len(t_obs) > 1:
-            observer_times = (
-                t_obs[:, np.newaxis] - arrival_times + np.min(arrival_times)
-            ).T
-        else:
-            observer_times = (t_obs - arrival_times + np.min(arrival_times))[
-                :, np.newaxis
-            ]
-
-        return observer_times
+        observer_times_list = []
+        for point_source_arrival_time in self.point_source_arrival_times():
```

Comment:
just making sure: this routine is not yet implemented for multiple source redshift, hence this function only works in limited ways. Perhaps add a TODO or warning to make sure users are not (yet) using it for multiple source planes